docs/how-to/deploy_sdcore_cups.md

```diff
@@ -8,70 +8,237 @@ This guide covers how to install a SD-Core 5G core network with Control Plane an
 
 ```console
-cat << EOF > upf-overlay.yaml
-applications:
-  upf:
-    options:
-      access-gateway-ip: 10.202.0.1
-      access-interface: access
-      access-ip: 10.202.0.10/24
-      core-gateway-ip: 10.203.0.1
-      core-interface: core
-      core-ip: 10.203.0.10/24
-      gnb-subnet: 10.204.0.0/24
-EOF
+resource "juju_offer" "amf-fiveg-n2" {
+  model            = "control-plane"
+  application_name = module.<CONTROL PLANE MODULE NAME>.amf_app_name
+  endpoint         = module.<CONTROL PLANE MODULE NAME>.fiveg_n2_endpoint
+}
+
+resource "juju_integration" "fiveg-n2" {
+  model = "control-plane"
+
+  application {
+    name     = <THE `fiveg-n2` REQUIRER APP>
+    endpoint = <THE `fiveg-n2` REQUIRER APP'S INTEGRATION ENDPOINT>
+  }
+
+  application {
+    offer_url = juju_offer.amf-fiveg-n2.url
+  }
+}
+```
+
+Apply the changes:
+
+```console
+terraform apply -auto-approve
 ```
 
+`````
+
+``````
+
+## Deploy SD-Core User Plane
+
 Create a Juju model.
 
 ```console
 juju add-model user-plane user-plane-cloud
 ```
 
-Deploy the user plane bundle.
+Get Charmed 5G Terraform modules by cloning the [Charmed 5G Terraform modules][Charmed 5G Terraform modules] Git repository.
+Inside the `modules/sdcore-user-plane-k8s` directory, create a `user-plane.tfvars` file to set the name of Juju model for the deployment:
+
+```console
+git clone https://github.com/canonical/terraform-juju-sdcore-k8s.git
+cd terraform-juju-sdcore-k8s/modules/sdcore-user-plane-k8s
+cat << EOF > user-plane.tfvars
+model_name = "user-plane"
+create_model = false
+upf_config = {
+  cni-type          = "macvlan"
+  access-gateway-ip = "10.202.0.1"
+  access-interface  = "access"
+  access-ip         = "10.202.0.10/24"
+  core-gateway-ip   = "10.203.0.1"
+  core-interface    = "core"
+  core-ip           = "10.203.0.10/24"
+  gnb-subnet        = "10.204.0.0/24"
+}
+
+EOF
+```
+
+Initialize Juju Terraform provider:
+
+```console
+terraform init
+```
+
+Deploy SD-Core User Plane
+
+```console
+terraform apply -var-file="user-plane.tfvars" -auto-approve
+```
+
+### Integration with the UPF N4 interface
+
+The UPF charm allows establishing the N4-plane connectivity through the `fiveg_n4` charm interface.
+
+``````{tab-set}
+
+`````{tab-item} Option 1: Integration within the same Juju model
+
+It is assumed that the `fiveg_n4` requirer application is already deployed in the Juju model.
+To create a `fiveg_n4` integration between the UPF and another application within the same Juju model, add the following section to the `main.tf` file in the `terraform-juju-sdcore-k8s/modules/sdcore-user-plane-k8s` directory:
+
+```console
+resource "juju_integration" "fiveg-n4" {
+  model = "user-plane"
+
+  application {
+    name     = module.upf.app_name
+    endpoint = module.upf.fiveg_n4_endpoint
+  }
+
+  application {
+    name     = <THE `fiveg_n4` REQUIRER APP>
```

Comment:
In general, fiveg_n4 requirer could be an external app. However we already deployed one in our sdcore-control plane which is `NMS`. Could we mention from our `NMS` as a sample requirer app which could be integrated as a preference.

Reply:
Similarly as above, I don't think that the how-to guide is the right place for this kind of examples.